test_stderr fails because stderr is "sh: sysctl command not found" when it should be empty

This is almost surely something I'm doing wrong or something weird about OSX.

* marbles.core version: 0.9.2
* marbles.mixins version: 0.9.2
* Python version: Python 3.6.5
* Operating System: Darwin jane.home 17.0.0 Darwin Kernel Version 17.0.0: Thu Aug 24 21:48:19 PDT 2017; root:xnu-4570.1.46~2/RELEASE_X86_64 x86_64

### Description

I set up a new development environment using pipenv and ran the tests to make sure that everything was looking good. It mostly is, except the `marbles.core.tests.test_main.VersionTestCase.test_stderr` test fails:

```
(marbles-rOKSo-69) jane@18:47:44 marbles (master %) $ tox -e coverage
... # full traceback included below
======================================================================
FAIL: test_stderr (tests.test_main.VersionTestCase)
The error output should be empty.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/jane/Development/marbles/marbles/core/tests/test_main.py", line 254, in test_stderr
    self.assertEqual('', self.stderr)
AssertionError: '' != 'sh: sysctl: command not found\n'
+ sh: sysctl: command not found
----------------------------------------------------------------------
Ran 230 tests in 14.262s

FAILED (failures=1, skipped=4)
Test failed: <unittest.runner.TextTestResult run=230 errors=0 failures=1>
error: Test failed: <unittest.runner.TextTestResult run=230 errors=0 failures=1>
ERROR: InvocationError for command '/Users/jane/Development/marbles/.tox/coverage/bin/pipenv run python -m coverage run marbles/core/setup.py test' (exited with code 1)
____________________________________________________________ summary _____________________________________________________________
ERROR:   coverage: commands failed
```

sysctl does appear to be on my path
```
(marbles-rOKSo-69) jane@18:47:44 marbles (master %) $ which sysctl
/usr/sbin/sysctl
```

### What I Did

```
(marbles-rOKSo-69) jane@18:46:41 marbles (master) $ tox -e coverage
GLOB sdist-make: /Users/jane/Development/marbles/setup.py
coverage installed: -i https://pypi.org/simple,-e ./marbles/core[dev],-e ./marbles/mixins[dev],coverage==4.5.1,flake8-per-file-ignores==0.6,flake8==3.5.0,mccabe==0.6.1,numpy==1.14.3,pandas==0.20.3,pathmatch==0.2.1,pycodestyle==2.3.1,pyflakes==1.6.0,python-dateutil==2.7.3,pytz==2018.4,six==1.11.0,typing==3.6.4
coverage runtests: PYTHONHASHSEED='589084020'
coverage runtests: commands[0] | pipenv install --dev
Courtesy Notice: Pipenv found itself running within a virtual environment, so it will automatically use that environment, instead of creating its own for any project. You can set PIPENV_IGNORE_VIRTUALENVS=1 to force pipenv to ignore that environment and create its own instead.
Installing dependencies from Pipfile.lock (fa7917)…
  🐍   ▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉ 52/52 — 00:00:06
coverage runtests: commands[1] | pipenv run python -m coverage erase
Courtesy Notice: Pipenv found itself running within a virtual environment, so it will automatically use that environment, instead of creating its own for any project. You can set PIPENV_IGNORE_VIRTUALENVS=1 to force pipenv to ignore that environment and create its own instead.
coverage runtests: commands[2] | pipenv run python -m coverage run marbles/core/setup.py test
Courtesy Notice: Pipenv found itself running within a virtual environment, so it will automatically use that environment, instead of creating its own for any project. You can set PIPENV_IGNORE_VIRTUALENVS=1 to force pipenv to ignore that environment and create its own instead.
running test
running egg_info
writing marbles.core.egg-info/PKG-INFO
writing dependency_links to marbles.core.egg-info/dependency_links.txt
writing namespace_packages to marbles.core.egg-info/namespace_packages.txt
writing requirements to marbles.core.egg-info/requires.txt
writing top-level names to marbles.core.egg-info/top_level.txt
package init file 'marbles/__init__.py' not found (or not a regular file)
package init file 'marbles/core/__init__.py' not found (or not a regular file)
reading manifest file 'marbles.core.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
writing manifest file 'marbles.core.egg-info/SOURCES.txt'
running build_ext
test_success (tests.test_log.TestAssertionLogging) (use_env=False, use_file=False, use_annotated_test_case=True)
On a successful assertion, do we log information? ... ok
test_capture_test_case_attributes (tests.test_log.TestAssertionLoggingAttributeCapture) (use_env=False, use_file=False, use_annotated_test_case=True)
Can we capture other attributes of a TestCase? ... ok
test_capture_test_case_attributes_on_failure (tests.test_log.TestAssertionLoggingAttributeCapture) (use_env=False, use_file=False, use_annotated_test_case=True)
Can we capture other attributes of a TestCase on failure? ... ok
test_success (tests.test_log.TestAssertionLoggingRespectsEnvOverrides) (use_env=False, use_file=False, use_annotated_test_case=True)
On a successful assertion, do we log information? ... ok
test_verbose_override (tests.test_log.TestAssertionLoggingRespectsEnvOverrides) (use_env=False, use_file=False, use_annotated_test_case=True)
On a successful assertion, do we log information? ... ok
test_capture_test_case_attributes (tests.test_log.TestAssertionLoggingVerboseAttributeCapture) (use_env=False, use_file=False, use_annotated_test_case=True)
Can we capture other attributes of a TestCase? ... ok
test_capture_test_case_attributes_on_failure (tests.test_log.TestAssertionLoggingVerboseAttributeCapture) (use_env=False, use_file=False, use_annotated_test_case=True)
Can we capture other attributes of a TestCase on failure? ... ok
test_failure (tests.test_log.TestAssertionLoggingVerboseFalse) (use_env=False, use_file=False, use_annotated_test_case=True)
On a failed assertion, do we log information? ... ok
test_success (tests.test_log.TestAssertionLoggingVerboseFalse) (use_env=False, use_file=False, use_annotated_test_case=True)
On a successful assertion, do we respect verbose=False? ... ok
test_failure (tests.test_log.TestAssertionLoggingVerboseList) (use_env=False, use_file=False, use_annotated_test_case=True)
On a failed assertion, do we log information? ... ok
test_success (tests.test_log.TestAssertionLoggingVerboseList) (use_env=False, use_file=False, use_annotated_test_case=True)
On a successful assertion, do we respect a verbose list? ... ok
test_failure (tests.test_log.TestAssertionLoggingVerboseTrue) (use_env=False, use_file=False, use_annotated_test_case=True)
On a failed assertion, do we log information? ... ok
test_success (tests.test_log.TestAssertionLoggingVerboseTrue) (use_env=False, use_file=False, use_annotated_test_case=True)
On a successful assertion, do we respect verbose=True? ... ok
test_success (tests.test_log.TestAssertionLogging) (use_env=False, use_file=True, use_annotated_test_case=True)
On a successful assertion, do we log information? ... ok
test_capture_test_case_attributes (tests.test_log.TestAssertionLoggingAttributeCapture) (use_env=False, use_file=True, use_annotated_test_case=True)
Can we capture other attributes of a TestCase? ... ok
test_capture_test_case_attributes_on_failure (tests.test_log.TestAssertionLoggingAttributeCapture) (use_env=False, use_file=True, use_annotated_test_case=True)
Can we capture other attributes of a TestCase on failure? ... ok
test_success (tests.test_log.TestAssertionLoggingRespectsEnvOverrides) (use_env=False, use_file=True, use_annotated_test_case=True)
On a successful assertion, do we log information? ... ok
test_verbose_override (tests.test_log.TestAssertionLoggingRespectsEnvOverrides) (use_env=False, use_file=True, use_annotated_test_case=True)
On a successful assertion, do we log information? ... ok
test_capture_test_case_attributes (tests.test_log.TestAssertionLoggingVerboseAttributeCapture) (use_env=False, use_file=True, use_annotated_test_case=True)
Can we capture other attributes of a TestCase? ... ok
test_capture_test_case_attributes_on_failure (tests.test_log.TestAssertionLoggingVerboseAttributeCapture) (use_env=False, use_file=True, use_annotated_test_case=True)
Can we capture other attributes of a TestCase on failure? ... ok
test_failure (tests.test_log.TestAssertionLoggingVerboseFalse) (use_env=False, use_file=True, use_annotated_test_case=True)
On a failed assertion, do we log information? ... ok
test_success (tests.test_log.TestAssertionLoggingVerboseFalse) (use_env=False, use_file=True, use_annotated_test_case=True)
On a successful assertion, do we respect verbose=False? ... ok
test_failure (tests.test_log.TestAssertionLoggingVerboseList) (use_env=False, use_file=True, use_annotated_test_case=True)
On a failed assertion, do we log information? ... ok
test_success (tests.test_log.TestAssertionLoggingVerboseList) (use_env=False, use_file=True, use_annotated_test_case=True)
On a successful assertion, do we respect a verbose list? ... ok
test_failure (tests.test_log.TestAssertionLoggingVerboseTrue) (use_env=False, use_file=True, use_annotated_test_case=True)
On a failed assertion, do we log information? ... ok
test_success (tests.test_log.TestAssertionLoggingVerboseTrue) (use_env=False, use_file=True, use_annotated_test_case=True)
On a successful assertion, do we respect verbose=True? ... ok
test_success (tests.test_log.TestAssertionLogging) (use_env=True, use_file=True, use_annotated_test_case=True)
On a successful assertion, do we log information? ... ok
test_capture_test_case_attributes (tests.test_log.TestAssertionLoggingAttributeCapture) (use_env=True, use_file=True, use_annotated_test_case=True)
Can we capture other attributes of a TestCase? ... ok
test_capture_test_case_attributes_on_failure (tests.test_log.TestAssertionLoggingAttributeCapture) (use_env=True, use_file=True, use_annotated_test_case=True)
Can we capture other attributes of a TestCase on failure? ... ok
test_success (tests.test_log.TestAssertionLoggingRespectsEnvOverrides) (use_env=True, use_file=True, use_annotated_test_case=True)
On a successful assertion, do we log information? ... skipped 'Only testing when the base class sets up with configure()'
test_verbose_override (tests.test_log.TestAssertionLoggingRespectsEnvOverrides) (use_env=True, use_file=True, use_annotated_test_case=True)
On a successful assertion, do we log information? ... skipped 'Only testing when the base class sets up with configure()'
test_capture_test_case_attributes (tests.test_log.TestAssertionLoggingVerboseAttributeCapture) (use_env=True, use_file=True, use_annotated_test_case=True)
Can we capture other attributes of a TestCase? ... ok
test_capture_test_case_attributes_on_failure (tests.test_log.TestAssertionLoggingVerboseAttributeCapture) (use_env=True, use_file=True, use_annotated_test_case=True)
Can we capture other attributes of a TestCase on failure? ... ok
test_failure (tests.test_log.TestAssertionLoggingVerboseFalse) (use_env=True, use_file=True, use_annotated_test_case=True)
On a failed assertion, do we log information? ... ok
test_success (tests.test_log.TestAssertionLoggingVerboseFalse) (use_env=True, use_file=True, use_annotated_test_case=True)
On a successful assertion, do we respect verbose=False? ... ok
test_failure (tests.test_log.TestAssertionLoggingVerboseList) (use_env=True, use_file=True, use_annotated_test_case=True)
On a failed assertion, do we log information? ... ok
test_success (tests.test_log.TestAssertionLoggingVerboseList) (use_env=True, use_file=True, use_annotated_test_case=True)
On a successful assertion, do we respect a verbose list? ... ok
test_failure (tests.test_log.TestAssertionLoggingVerboseTrue) (use_env=True, use_file=True, use_annotated_test_case=True)
On a failed assertion, do we log information? ... ok
test_success (tests.test_log.TestAssertionLoggingVerboseTrue) (use_env=True, use_file=True, use_annotated_test_case=True)
On a successful assertion, do we respect verbose=True? ... ok
test_success (tests.test_log.TestAssertionLogging) (use_env=False, use_file=False, use_annotated_test_case=False)
On a successful assertion, do we log information? ... ok
test_capture_test_case_attributes (tests.test_log.TestAssertionLoggingAttributeCapture) (use_env=False, use_file=False, use_annotated_test_case=False)
Can we capture other attributes of a TestCase? ... ok
test_capture_test_case_attributes_on_failure (tests.test_log.TestAssertionLoggingAttributeCapture) (use_env=False, use_file=False, use_annotated_test_case=False)
Can we capture other attributes of a TestCase on failure? ... ok
test_success (tests.test_log.TestAssertionLoggingRespectsEnvOverrides) (use_env=False, use_file=False, use_annotated_test_case=False)
On a successful assertion, do we log information? ... ok
test_verbose_override (tests.test_log.TestAssertionLoggingRespectsEnvOverrides) (use_env=False, use_file=False, use_annotated_test_case=False)
On a successful assertion, do we log information? ... ok
test_capture_test_case_attributes (tests.test_log.TestAssertionLoggingVerboseAttributeCapture) (use_env=False, use_file=False, use_annotated_test_case=False)
Can we capture other attributes of a TestCase? ... ok
test_capture_test_case_attributes_on_failure (tests.test_log.TestAssertionLoggingVerboseAttributeCapture) (use_env=False, use_file=False, use_annotated_test_case=False)
Can we capture other attributes of a TestCase on failure? ... ok
test_failure (tests.test_log.TestAssertionLoggingVerboseFalse) (use_env=False, use_file=False, use_annotated_test_case=False)
On a failed assertion, do we log information? ... ok
test_success (tests.test_log.TestAssertionLoggingVerboseFalse) (use_env=False, use_file=False, use_annotated_test_case=False)
On a successful assertion, do we respect verbose=False? ... ok
test_failure (tests.test_log.TestAssertionLoggingVerboseList) (use_env=False, use_file=False, use_annotated_test_case=False)
On a failed assertion, do we log information? ... ok
test_success (tests.test_log.TestAssertionLoggingVerboseList) (use_env=False, use_file=False, use_annotated_test_case=False)
On a successful assertion, do we respect a verbose list? ... ok
test_failure (tests.test_log.TestAssertionLoggingVerboseTrue) (use_env=False, use_file=False, use_annotated_test_case=False)
On a failed assertion, do we log information? ... ok
test_success (tests.test_log.TestAssertionLoggingVerboseTrue) (use_env=False, use_file=False, use_annotated_test_case=False)
On a successful assertion, do we respect verbose=True? ... ok
test_success (tests.test_log.TestAssertionLogging) (use_env=False, use_file=True, use_annotated_test_case=False)
On a successful assertion, do we log information? ... ok
test_capture_test_case_attributes (tests.test_log.TestAssertionLoggingAttributeCapture) (use_env=False, use_file=True, use_annotated_test_case=False)
Can we capture other attributes of a TestCase? ... ok
test_capture_test_case_attributes_on_failure (tests.test_log.TestAssertionLoggingAttributeCapture) (use_env=False, use_file=True, use_annotated_test_case=False)
Can we capture other attributes of a TestCase on failure? ... ok
test_success (tests.test_log.TestAssertionLoggingRespectsEnvOverrides) (use_env=False, use_file=True, use_annotated_test_case=False)
On a successful assertion, do we log information? ... ok
test_verbose_override (tests.test_log.TestAssertionLoggingRespectsEnvOverrides) (use_env=False, use_file=True, use_annotated_test_case=False)
On a successful assertion, do we log information? ... ok
test_capture_test_case_attributes (tests.test_log.TestAssertionLoggingVerboseAttributeCapture) (use_env=False, use_file=True, use_annotated_test_case=False)
Can we capture other attributes of a TestCase? ... ok
test_capture_test_case_attributes_on_failure (tests.test_log.TestAssertionLoggingVerboseAttributeCapture) (use_env=False, use_file=True, use_annotated_test_case=False)
Can we capture other attributes of a TestCase on failure? ... ok
test_failure (tests.test_log.TestAssertionLoggingVerboseFalse) (use_env=False, use_file=True, use_annotated_test_case=False)
On a failed assertion, do we log information? ... ok
test_success (tests.test_log.TestAssertionLoggingVerboseFalse) (use_env=False, use_file=True, use_annotated_test_case=False)
On a successful assertion, do we respect verbose=False? ... ok
test_failure (tests.test_log.TestAssertionLoggingVerboseList) (use_env=False, use_file=True, use_annotated_test_case=False)
On a failed assertion, do we log information? ... ok
test_success (tests.test_log.TestAssertionLoggingVerboseList) (use_env=False, use_file=True, use_annotated_test_case=False)
On a successful assertion, do we respect a verbose list? ... ok
test_failure (tests.test_log.TestAssertionLoggingVerboseTrue) (use_env=False, use_file=True, use_annotated_test_case=False)
On a failed assertion, do we log information? ... ok
test_success (tests.test_log.TestAssertionLoggingVerboseTrue) (use_env=False, use_file=True, use_annotated_test_case=False)
On a successful assertion, do we respect verbose=True? ... ok
test_success (tests.test_log.TestAssertionLogging) (use_env=True, use_file=True, use_annotated_test_case=False)
On a successful assertion, do we log information? ... ok
test_capture_test_case_attributes (tests.test_log.TestAssertionLoggingAttributeCapture) (use_env=True, use_file=True, use_annotated_test_case=False)
Can we capture other attributes of a TestCase? ... ok
test_capture_test_case_attributes_on_failure (tests.test_log.TestAssertionLoggingAttributeCapture) (use_env=True, use_file=True, use_annotated_test_case=False)
Can we capture other attributes of a TestCase on failure? ... ok
test_success (tests.test_log.TestAssertionLoggingRespectsEnvOverrides) (use_env=True, use_file=True, use_annotated_test_case=False)
On a successful assertion, do we log information? ... skipped 'Only testing when the base class sets up with configure()'
test_verbose_override (tests.test_log.TestAssertionLoggingRespectsEnvOverrides) (use_env=True, use_file=True, use_annotated_test_case=False)
On a successful assertion, do we log information? ... skipped 'Only testing when the base class sets up with configure()'
test_capture_test_case_attributes (tests.test_log.TestAssertionLoggingVerboseAttributeCapture) (use_env=True, use_file=True, use_annotated_test_case=False)
Can we capture other attributes of a TestCase? ... ok
test_capture_test_case_attributes_on_failure (tests.test_log.TestAssertionLoggingVerboseAttributeCapture) (use_env=True, use_file=True, use_annotated_test_case=False)
Can we capture other attributes of a TestCase on failure? ... ok
test_failure (tests.test_log.TestAssertionLoggingVerboseFalse) (use_env=True, use_file=True, use_annotated_test_case=False)
On a failed assertion, do we log information? ... ok
test_success (tests.test_log.TestAssertionLoggingVerboseFalse) (use_env=True, use_file=True, use_annotated_test_case=False)
On a successful assertion, do we respect verbose=False? ... ok
test_failure (tests.test_log.TestAssertionLoggingVerboseList) (use_env=True, use_file=True, use_annotated_test_case=False)
On a failed assertion, do we log information? ... ok
test_success (tests.test_log.TestAssertionLoggingVerboseList) (use_env=True, use_file=True, use_annotated_test_case=False)
On a successful assertion, do we respect a verbose list? ... ok
test_failure (tests.test_log.TestAssertionLoggingVerboseTrue) (use_env=True, use_file=True, use_annotated_test_case=False)
On a failed assertion, do we log information? ... ok
test_success (tests.test_log.TestAssertionLoggingVerboseTrue) (use_env=True, use_file=True, use_annotated_test_case=False)
On a successful assertion, do we respect verbose=True? ... ok
test_annotated_assertion_error_not_raised (tests.test_marbles.InterfaceTestCase) (use_annotated_test_case=True)
Is no error raised if a test succeeds? ... ok
test_annotated_assertion_error_raised (tests.test_marbles.InterfaceTestCase) (use_annotated_test_case=True)
Is an ContextualAssertionError raised if a test fails? ... ok
test_assert_raises_failure (tests.test_marbles.InterfaceTestCase) (use_annotated_test_case=True)
Does assertRaises work correctly when the test fails? ... ok
test_assert_raises_missing_note (tests.test_marbles.InterfaceTestCase) (use_annotated_test_case=True)
Do we notice missing note for assertRaises? ... ok
test_assert_raises_success (tests.test_marbles.InterfaceTestCase) (use_annotated_test_case=True)
Does assertRaises work correctly when the test passes? ... ok
test_fail_handles_note_properly (tests.test_marbles.InterfaceTestCase) (use_annotated_test_case=True)
Does TestCase.fail() deal with note the right way? ... ok
test_fail_rejects_extra_args (tests.test_marbles.InterfaceTestCase) (use_annotated_test_case=True)
Does TestCase.fail() reject extra arguments? ... ok
test_fail_works_when_invoked_by_builtin_assertions (tests.test_marbles.InterfaceTestCase) (use_annotated_test_case=True) ... ok
test_missing_annotation (tests.test_marbles.InterfaceTestCase) (use_annotated_test_case=True)
Does marbles check for missing annotations? ... ok
test_missing_msg_ok (tests.test_marbles.InterfaceTestCase) (use_annotated_test_case=True)
Is it ok to provide only note? ... ok
test_missing_note_dict (tests.test_marbles.InterfaceTestCase) (use_annotated_test_case=True)
When passing a dict as msg, do we still check for note? ... ok
test_odd_argument_order_ok (tests.test_marbles.InterfaceTestCase) (use_annotated_test_case=True)
Does marbles handle a msg argument before the last position? ... ok
test_string_equality (tests.test_marbles.InterfaceTestCase) (use_annotated_test_case=True)
Can we use assertEqual on strings? ... ok
test_success (tests.test_marbles.TestAssertionLoggingFailure) (use_annotated_test_case=True)
When logging fails, do we allow the test to proceed? ... ok
test_assert_raises_kwargs_msg (tests.test_marbles.TestContextualAssertionError) (use_annotated_test_case=True)
Do we capture kwargs annotations properly for assertRaises? ... ok
test_assert_raises_without_msg (tests.test_marbles.TestContextualAssertionError) (use_annotated_test_case=True)
Do we capture annotations properly for assertRaises? ... ok
test_assert_stmt_indicates_line (tests.test_marbles.TestContextualAssertionError) (use_annotated_test_case=True)
Does e.assert_stmt indicate the line from the source code? ... ok
test_assert_stmt_surrounding_lines (tests.test_marbles.TestContextualAssertionError) (use_annotated_test_case=True)
Does _find_assert_stmt read surrounding lines from the file? ... ok
test_custom_assertions (tests.test_marbles.TestContextualAssertionError) (use_annotated_test_case=True)
Does the marbles note work with custom-defined assertions? ... ok
test_custom_assertions_kwargs (tests.test_marbles.TestContextualAssertionError) (use_annotated_test_case=True)
Does the marbles kwargs note work with custom assertions? ... ok
test_exclude_ignored_locals (tests.test_marbles.TestContextualAssertionError) (use_annotated_test_case=True)
Are ignored variables excluded from output? ... ok
test_exclude_internal_mangled_locals (tests.test_marbles.TestContextualAssertionError) (use_annotated_test_case=True)
Are internal/mangled variables excluded from the "Locals"? ... ok
test_get_stack (tests.test_marbles.TestContextualAssertionError) (use_annotated_test_case=True)
Does _get_stack() find the stack level with the test definition? ... ok
test_kwargs_stick_together (tests.test_marbles.TestContextualAssertionError) (use_annotated_test_case=True)
Does the kwargs form of an assertion enforce that message and ... ok
test_locals_hidden_when_all_private (tests.test_marbles.TestContextualAssertionError) (use_annotated_test_case=True)
Does marbles hide the Locals section if all are private? ... ok
test_locals_hidden_when_missing (tests.test_marbles.TestContextualAssertionError) (use_annotated_test_case=True)
Does marbles hide the Locals section if there are none? ... ok
test_locals_shown_when_present (tests.test_marbles.TestContextualAssertionError) (use_annotated_test_case=True)
Does marbles show the Locals section if there are some? ... ok
test_missing_msg_dict (tests.test_marbles.TestContextualAssertionError) (use_annotated_test_case=True)
Is the default msg properly displayed when note is in a dict? ... ok
test_missing_msg_kwargs_note (tests.test_marbles.TestContextualAssertionError) (use_annotated_test_case=True)
Is the default msg properly displayed? ... ok
test_named_assert_args (tests.test_marbles.TestContextualAssertionError) (use_annotated_test_case=True)
Is annotation captured correctly if named arguments are provided? ... ok
test_note_rich_format_strings (tests.test_marbles.TestContextualAssertionError) (use_annotated_test_case=True) ... ok
test_note_wrapping (tests.test_marbles.TestContextualAssertionError) (use_annotated_test_case=True)
Do we wrap the note properly? ... ok
test_odd_argument_order (tests.test_marbles.TestContextualAssertionError) (use_annotated_test_case=True)
Does marbles handle a msg argument before the last position? ... ok
test_positional_assert_args (tests.test_marbles.TestContextualAssertionError) (use_annotated_test_case=True)
Is annotation captured correctly when using positional arguments? ... ok
test_positional_msg_kwargs_note (tests.test_marbles.TestContextualAssertionError) (use_annotated_test_case=True)
Is annotation captured correctly when using a positional msg? ... ok
test_use_kwargs_form (tests.test_marbles.TestContextualAssertionError) (use_annotated_test_case=True)
Does the kwargs form of an assertion work? ... ok
test_verify_annotation_dict_missing_keys (tests.test_marbles.TestContextualAssertionError) (use_annotated_test_case=True)
Is an Exception raised if annotation is missing expected keys? ... ok
test_verify_annotation_locals (tests.test_marbles.TestContextualAssertionError) (use_annotated_test_case=True)
Are locals in the test definition formatted into annotations? ... ok
test_verify_annotation_none (tests.test_marbles.TestContextualAssertionError) (use_annotated_test_case=True)
Is an Exception raised if no annotation is provided? ... ok
test_annotated_assertion_error_not_raised (tests.test_marbles.InterfaceTestCase) (use_annotated_test_case=False)
Is no error raised if a test succeeds? ... ok
test_annotated_assertion_error_raised (tests.test_marbles.InterfaceTestCase) (use_annotated_test_case=False)
Is an ContextualAssertionError raised if a test fails? ... ok
test_assert_raises_failure (tests.test_marbles.InterfaceTestCase) (use_annotated_test_case=False)
Does assertRaises work correctly when the test fails? ... ok
test_assert_raises_missing_note (tests.test_marbles.InterfaceTestCase) (use_annotated_test_case=False)
Do we notice missing note for assertRaises? ... ok
test_assert_raises_success (tests.test_marbles.InterfaceTestCase) (use_annotated_test_case=False)
Does assertRaises work correctly when the test passes? ... ok
test_fail_handles_note_properly (tests.test_marbles.InterfaceTestCase) (use_annotated_test_case=False)
Does TestCase.fail() deal with note the right way? ... ok
test_fail_rejects_extra_args (tests.test_marbles.InterfaceTestCase) (use_annotated_test_case=False)
Does TestCase.fail() reject extra arguments? ... ok
test_fail_works_when_invoked_by_builtin_assertions (tests.test_marbles.InterfaceTestCase) (use_annotated_test_case=False) ... ok
test_missing_annotation (tests.test_marbles.InterfaceTestCase) (use_annotated_test_case=False)
Does marbles check for missing annotations? ... ok
test_missing_msg_ok (tests.test_marbles.InterfaceTestCase) (use_annotated_test_case=False)
Is it ok to provide only note? ... ok
test_missing_note_dict (tests.test_marbles.InterfaceTestCase) (use_annotated_test_case=False)
When passing a dict as msg, do we still check for note? ... ok
test_odd_argument_order_ok (tests.test_marbles.InterfaceTestCase) (use_annotated_test_case=False)
Does marbles handle a msg argument before the last position? ... ok
test_string_equality (tests.test_marbles.InterfaceTestCase) (use_annotated_test_case=False)
Can we use assertEqual on strings? ... ok
test_success (tests.test_marbles.TestAssertionLoggingFailure) (use_annotated_test_case=False)
When logging fails, do we allow the test to proceed? ... ok
test_assert_raises_kwargs_msg (tests.test_marbles.TestContextualAssertionError) (use_annotated_test_case=False)
Do we capture kwargs annotations properly for assertRaises? ... ok
test_assert_raises_without_msg (tests.test_marbles.TestContextualAssertionError) (use_annotated_test_case=False)
Do we capture annotations properly for assertRaises? ... ok
test_assert_stmt_indicates_line (tests.test_marbles.TestContextualAssertionError) (use_annotated_test_case=False)
Does e.assert_stmt indicate the line from the source code? ... ok
test_assert_stmt_surrounding_lines (tests.test_marbles.TestContextualAssertionError) (use_annotated_test_case=False)
Does _find_assert_stmt read surrounding lines from the file? ... ok
test_custom_assertions (tests.test_marbles.TestContextualAssertionError) (use_annotated_test_case=False)
Does the marbles note work with custom-defined assertions? ... ok
test_custom_assertions_kwargs (tests.test_marbles.TestContextualAssertionError) (use_annotated_test_case=False)
Does the marbles kwargs note work with custom assertions? ... ok
test_exclude_ignored_locals (tests.test_marbles.TestContextualAssertionError) (use_annotated_test_case=False)
Are ignored variables excluded from output? ... ok
test_exclude_internal_mangled_locals (tests.test_marbles.TestContextualAssertionError) (use_annotated_test_case=False)
Are internal/mangled variables excluded from the "Locals"? ... ok
test_get_stack (tests.test_marbles.TestContextualAssertionError) (use_annotated_test_case=False)
Does _get_stack() find the stack level with the test definition? ... ok
test_kwargs_stick_together (tests.test_marbles.TestContextualAssertionError) (use_annotated_test_case=False)
Does the kwargs form of an assertion enforce that message and ... ok
test_locals_hidden_when_all_private (tests.test_marbles.TestContextualAssertionError) (use_annotated_test_case=False)
Does marbles hide the Locals section if all are private? ... ok
test_locals_hidden_when_missing (tests.test_marbles.TestContextualAssertionError) (use_annotated_test_case=False)
Does marbles hide the Locals section if there are none? ... ok
test_locals_shown_when_present (tests.test_marbles.TestContextualAssertionError) (use_annotated_test_case=False)
Does marbles show the Locals section if there are some? ... ok
test_missing_msg_dict (tests.test_marbles.TestContextualAssertionError) (use_annotated_test_case=False)
Is the default msg properly displayed when note is in a dict? ... ok
test_missing_msg_kwargs_note (tests.test_marbles.TestContextualAssertionError) (use_annotated_test_case=False)
Is the default msg properly displayed? ... ok
test_named_assert_args (tests.test_marbles.TestContextualAssertionError) (use_annotated_test_case=False)
Is annotation captured correctly if named arguments are provided? ... ok
test_note_rich_format_strings (tests.test_marbles.TestContextualAssertionError) (use_annotated_test_case=False) ... ok
test_note_wrapping (tests.test_marbles.TestContextualAssertionError) (use_annotated_test_case=False)
Do we wrap the note properly? ... ok
test_odd_argument_order (tests.test_marbles.TestContextualAssertionError) (use_annotated_test_case=False)
Does marbles handle a msg argument before the last position? ... ok
test_positional_assert_args (tests.test_marbles.TestContextualAssertionError) (use_annotated_test_case=False)
Is annotation captured correctly when using positional arguments? ... ok
test_positional_msg_kwargs_note (tests.test_marbles.TestContextualAssertionError) (use_annotated_test_case=False)
Is annotation captured correctly when using a positional msg? ... ok
test_use_kwargs_form (tests.test_marbles.TestContextualAssertionError) (use_annotated_test_case=False)
Does the kwargs form of an assertion work? ... ok
test_verify_annotation_dict_missing_keys (tests.test_marbles.TestContextualAssertionError) (use_annotated_test_case=False)
Is an Exception raised if annotation is missing expected keys? ... ok
test_verify_annotation_locals (tests.test_marbles.TestContextualAssertionError) (use_annotated_test_case=False)
Are locals in the test definition formatted into annotations? ... ok
test_verify_annotation_none (tests.test_marbles.TestContextualAssertionError) (use_annotated_test_case=False)
Is an Exception raised if no annotation is provided? ... ok
test_empty_stdout (tests.test_main.MainWithFailureTestCase) (runner='unittest', verbose=True, test_file='example_unittest.py')
Standard out should be empty in all cases. ... ok
test_show_locals (tests.test_main.MainWithFailureTestCase) (runner='unittest', verbose=True, test_file='example_unittest.py')
Locals should be printed. ... ok
test_show_msg (tests.test_main.MainWithFailureTestCase) (runner='unittest', verbose=True, test_file='example_unittest.py')
The failure message should always appear. ... ok
test_show_source (tests.test_main.MainWithFailureTestCase) (runner='unittest', verbose=True, test_file='example_unittest.py')
The source code should appear. ... ok
test_traceback (tests.test_main.MainWithFailureTestCase) (runner='unittest', verbose=True, test_file='example_unittest.py')
The traceback should only be shown in verbose mode. ... ok
test_empty_stdout (tests.test_main.MainWithFailureTestCase) (runner='unittest', verbose=True, test_file='example_marbles.py')
Standard out should be empty in all cases. ... ok
test_show_locals (tests.test_main.MainWithFailureTestCase) (runner='unittest', verbose=True, test_file='example_marbles.py')
Locals should be printed. ... ok
test_show_msg (tests.test_main.MainWithFailureTestCase) (runner='unittest', verbose=True, test_file='example_marbles.py')
The failure message should always appear. ... ok
test_show_source (tests.test_main.MainWithFailureTestCase) (runner='unittest', verbose=True, test_file='example_marbles.py')
The source code should appear. ... ok
test_traceback (tests.test_main.MainWithFailureTestCase) (runner='unittest', verbose=True, test_file='example_marbles.py')
The traceback should only be shown in verbose mode. ... ok
test_empty_stdout (tests.test_main.MainWithFailureTestCase) (runner='unittest', verbose=False, test_file='example_unittest.py')
Standard out should be empty in all cases. ... ok
test_show_locals (tests.test_main.MainWithFailureTestCase) (runner='unittest', verbose=False, test_file='example_unittest.py')
Locals should be printed. ... ok
test_show_msg (tests.test_main.MainWithFailureTestCase) (runner='unittest', verbose=False, test_file='example_unittest.py')
The failure message should always appear. ... ok
test_show_source (tests.test_main.MainWithFailureTestCase) (runner='unittest', verbose=False, test_file='example_unittest.py')
The source code should appear. ... ok
test_traceback (tests.test_main.MainWithFailureTestCase) (runner='unittest', verbose=False, test_file='example_unittest.py')
The traceback should only be shown in verbose mode. ... ok
test_empty_stdout (tests.test_main.MainWithFailureTestCase) (runner='unittest', verbose=False, test_file='example_marbles.py')
Standard out should be empty in all cases. ... ok
test_show_locals (tests.test_main.MainWithFailureTestCase) (runner='unittest', verbose=False, test_file='example_marbles.py')
Locals should be printed. ... ok
test_show_msg (tests.test_main.MainWithFailureTestCase) (runner='unittest', verbose=False, test_file='example_marbles.py')
The failure message should always appear. ... ok
test_show_source (tests.test_main.MainWithFailureTestCase) (runner='unittest', verbose=False, test_file='example_marbles.py')
The source code should appear. ... ok
test_traceback (tests.test_main.MainWithFailureTestCase) (runner='unittest', verbose=False, test_file='example_marbles.py')
The traceback should only be shown in verbose mode. ... ok
test_empty_stdout (tests.test_main.MainWithFailureTestCase) (runner='marbles', verbose=True, test_file='example_unittest.py')
Standard out should be empty in all cases. ... ok
test_show_locals (tests.test_main.MainWithFailureTestCase) (runner='marbles', verbose=True, test_file='example_unittest.py')
Locals should be printed. ... ok
test_show_msg (tests.test_main.MainWithFailureTestCase) (runner='marbles', verbose=True, test_file='example_unittest.py')
The failure message should always appear. ... ok
test_show_source (tests.test_main.MainWithFailureTestCase) (runner='marbles', verbose=True, test_file='example_unittest.py')
The source code should appear. ... ok
test_traceback (tests.test_main.MainWithFailureTestCase) (runner='marbles', verbose=True, test_file='example_unittest.py')
The traceback should only be shown in verbose mode. ... ok
test_empty_stdout (tests.test_main.MainWithFailureTestCase) (runner='marbles', verbose=True, test_file='example_marbles.py')
Standard out should be empty in all cases. ... ok
test_show_locals (tests.test_main.MainWithFailureTestCase) (runner='marbles', verbose=True, test_file='example_marbles.py')
Locals should be printed. ... ok
test_show_msg (tests.test_main.MainWithFailureTestCase) (runner='marbles', verbose=True, test_file='example_marbles.py')
The failure message should always appear. ... ok
test_show_source (tests.test_main.MainWithFailureTestCase) (runner='marbles', verbose=True, test_file='example_marbles.py')
The source code should appear. ... ok
test_traceback (tests.test_main.MainWithFailureTestCase) (runner='marbles', verbose=True, test_file='example_marbles.py')
The traceback should only be shown in verbose mode. ... ok
test_empty_stdout (tests.test_main.MainWithFailureTestCase) (runner='marbles', verbose=False, test_file='example_unittest.py')
Standard out should be empty in all cases. ... ok
test_show_locals (tests.test_main.MainWithFailureTestCase) (runner='marbles', verbose=False, test_file='example_unittest.py')
Locals should be printed. ... ok
test_show_msg (tests.test_main.MainWithFailureTestCase) (runner='marbles', verbose=False, test_file='example_unittest.py')
The failure message should always appear. ... ok
test_show_source (tests.test_main.MainWithFailureTestCase) (runner='marbles', verbose=False, test_file='example_unittest.py')
The source code should appear. ... ok
test_traceback (tests.test_main.MainWithFailureTestCase) (runner='marbles', verbose=False, test_file='example_unittest.py')
The traceback should only be shown in verbose mode. ... ok
test_empty_stdout (tests.test_main.MainWithFailureTestCase) (runner='marbles', verbose=False, test_file='example_marbles.py')
Standard out should be empty in all cases. ... ok
test_show_locals (tests.test_main.MainWithFailureTestCase) (runner='marbles', verbose=False, test_file='example_marbles.py')
Locals should be printed. ... ok
test_show_msg (tests.test_main.MainWithFailureTestCase) (runner='marbles', verbose=False, test_file='example_marbles.py')
The failure message should always appear. ... ok
test_show_source (tests.test_main.MainWithFailureTestCase) (runner='marbles', verbose=False, test_file='example_marbles.py')
The source code should appear. ... ok
test_traceback (tests.test_main.MainWithFailureTestCase) (runner='marbles', verbose=False, test_file='example_marbles.py')
The traceback should only be shown in verbose mode. ... ok
test_empty_stdout (tests.test_main.MainWithFailureTestCase) (runner='script', verbose=True, test_file='example_unittest.py')
Standard out should be empty in all cases. ... ok
test_show_locals (tests.test_main.MainWithFailureTestCase) (runner='script', verbose=True, test_file='example_unittest.py')
Locals should be printed. ... ok
test_show_msg (tests.test_main.MainWithFailureTestCase) (runner='script', verbose=True, test_file='example_unittest.py')
The failure message should always appear. ... ok
test_show_source (tests.test_main.MainWithFailureTestCase) (runner='script', verbose=True, test_file='example_unittest.py')
The source code should appear. ... ok
test_traceback (tests.test_main.MainWithFailureTestCase) (runner='script', verbose=True, test_file='example_unittest.py')
The traceback should only be shown in verbose mode. ... ok
test_empty_stdout (tests.test_main.MainWithFailureTestCase) (runner='script', verbose=True, test_file='example_marbles.py')
Standard out should be empty in all cases. ... ok
test_show_locals (tests.test_main.MainWithFailureTestCase) (runner='script', verbose=True, test_file='example_marbles.py')
Locals should be printed. ... ok
test_show_msg (tests.test_main.MainWithFailureTestCase) (runner='script', verbose=True, test_file='example_marbles.py')
The failure message should always appear. ... ok
test_show_source (tests.test_main.MainWithFailureTestCase) (runner='script', verbose=True, test_file='example_marbles.py')
The source code should appear. ... ok
test_traceback (tests.test_main.MainWithFailureTestCase) (runner='script', verbose=True, test_file='example_marbles.py')
The traceback should only be shown in verbose mode. ... ok
test_empty_stdout (tests.test_main.MainWithFailureTestCase) (runner='script', verbose=False, test_file='example_unittest.py')
Standard out should be empty in all cases. ... ok
test_show_locals (tests.test_main.MainWithFailureTestCase) (runner='script', verbose=False, test_file='example_unittest.py')
Locals should be printed. ... ok
test_show_msg (tests.test_main.MainWithFailureTestCase) (runner='script', verbose=False, test_file='example_unittest.py')
The failure message should always appear. ... ok
test_show_source (tests.test_main.MainWithFailureTestCase) (runner='script', verbose=False, test_file='example_unittest.py')
The source code should appear. ... ok
test_traceback (tests.test_main.MainWithFailureTestCase) (runner='script', verbose=False, test_file='example_unittest.py')
The traceback should only be shown in verbose mode. ... ok
test_empty_stdout (tests.test_main.MainWithFailureTestCase) (runner='script', verbose=False, test_file='example_marbles.py')
Standard out should be empty in all cases. ... ok
test_show_locals (tests.test_main.MainWithFailureTestCase) (runner='script', verbose=False, test_file='example_marbles.py')
Locals should be printed. ... ok
test_show_msg (tests.test_main.MainWithFailureTestCase) (runner='script', verbose=False, test_file='example_marbles.py')
The failure message should always appear. ... ok
test_show_source (tests.test_main.MainWithFailureTestCase) (runner='script', verbose=False, test_file='example_marbles.py')
The source code should appear. ... ok
test_traceback (tests.test_main.MainWithFailureTestCase) (runner='script', verbose=False, test_file='example_marbles.py')
The traceback should only be shown in verbose mode. ... ok
test_empty_stdout (tests.test_main.MainWithErrorTestCase) (runner='unittest', verbose=True, test_file='example_error.py')
Standard out should be empty in all cases. ... ok
test_traceback (tests.test_main.MainWithErrorTestCase) (runner='unittest', verbose=True, test_file='example_error.py')
The traceback should be shown for all errors. ... ok
test_empty_stdout (tests.test_main.MainWithErrorTestCase) (runner='unittest', verbose=False, test_file='example_error.py')
Standard out should be empty in all cases. ... ok
test_traceback (tests.test_main.MainWithErrorTestCase) (runner='unittest', verbose=False, test_file='example_error.py')
The traceback should be shown for all errors. ... ok
test_empty_stdout (tests.test_main.MainWithErrorTestCase) (runner='marbles', verbose=True, test_file='example_error.py')
Standard out should be empty in all cases. ... ok
test_traceback (tests.test_main.MainWithErrorTestCase) (runner='marbles', verbose=True, test_file='example_error.py')
The traceback should be shown for all errors. ... ok
test_empty_stdout (tests.test_main.MainWithErrorTestCase) (runner='marbles', verbose=False, test_file='example_error.py')
Standard out should be empty in all cases. ... ok
test_traceback (tests.test_main.MainWithErrorTestCase) (runner='marbles', verbose=False, test_file='example_error.py')
The traceback should be shown for all errors. ... ok
test_empty_stdout (tests.test_main.MainWithErrorTestCase) (runner='script', verbose=True, test_file='example_error.py')
Standard out should be empty in all cases. ... ok
test_traceback (tests.test_main.MainWithErrorTestCase) (runner='script', verbose=True, test_file='example_error.py')
The traceback should be shown for all errors. ... ok
test_empty_stdout (tests.test_main.MainWithErrorTestCase) (runner='script', verbose=False, test_file='example_error.py')
Standard out should be empty in all cases. ... ok
test_traceback (tests.test_main.MainWithErrorTestCase) (runner='script', verbose=False, test_file='example_error.py')
The traceback should be shown for all errors. ... ok
test_stderr (tests.test_main.VersionTestCase)
The error output should be empty. ... FAIL
test_stdout (tests.test_main.VersionTestCase)
The version output should contain marbles.core's version. ... ok

======================================================================
FAIL: test_stderr (tests.test_main.VersionTestCase)
The error output should be empty.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/jane/Development/marbles/marbles/core/tests/test_main.py", line 254, in test_stderr
    self.assertEqual('', self.stderr)
AssertionError: '' != 'sh: sysctl: command not found\n'
+ sh: sysctl: command not found


----------------------------------------------------------------------
Ran 230 tests in 14.262s

FAILED (failures=1, skipped=4)
Test failed: <unittest.runner.TextTestResult run=230 errors=0 failures=1>
error: Test failed: <unittest.runner.TextTestResult run=230 errors=0 failures=1>
ERROR: InvocationError for command '/Users/jane/Development/marbles/.tox/coverage/bin/pipenv run python -m coverage run marbles/core/setup.py test' (exited with code 1)
____________________________________________________________ summary _____________________________________________________________
ERROR:   coverage: commands failed
```

### Expected behavior

I expected all the tests to pass :(

### Additional context

The `test_stderr` test fails when I run the tests with the below commands as well
```
(marbles-rOKSo-69) jane@18:51:00 core (master) $ cd marbles/core/
(marbles-rOKSo-69) jane@18:51:00 core (master) $ coverage run setup.py test
(marbles-rOKSo-69) jane@18:51:00 core (master) $ python -m unittest
```

So I had that thought, but it appears not to be?

```
(marbles-rOKSo-69) jane@19:06:40 core (master *) $ python -m unittest tests.test_main.VersionTestCase.test_stderr
['/Users/jane/Environments/marbles-rOKSo-69/bin/python', '-m', 'marbles', '--version']
F
======================================================================
FAIL: test_stderr (tests.test_main.VersionTestCase)
The error output should be empty.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/jane/Development/marbles/marbles/core/tests/test_main.py", line 255, in test_stderr
    self.assertEqual('', self.stderr)
AssertionError: '' != 'sh: sysctl: command not found\n'
+ sh: sysctl: command not found


----------------------------------------------------------------------
Ran 1 test in 0.350s

FAILED (failures=1)
(marbles-rOKSo-69) jane@19:06:48 core (master *) $ git diff tests/test_main.py
diff --git a/marbles/core/tests/test_main.py b/marbles/core/tests/test_main.py
index 2b9deca..507a4ff 100644
--- a/marbles/core/tests/test_main.py
+++ b/marbles/core/tests/test_main.py
@@ -251,6 +251,7 @@ class VersionTestCase(CommandRunningTestCase):
 
     def test_stderr(self):
         '''The error output should be empty.'''
+        print(self.cmd)
         self.assertEqual('', self.stderr)
```

--- a/marbles/core/tests/test_main.py
+++ b/marbles/core/tests/test_main.py
@@ -79,10 +79,10 @@ class CommandRunningTestCase(unittest.TestCase):
                 pth.flush()
 
             coverage_config = os.path.join(core_dir, '.coveragerc')
-            env = {'COVERAGE_PROCESS_START': coverage_config}
+            env = {**os.environ, **{'COVERAGE_PROCESS_START': coverage_config}}
             to_remove = pth.name
         else:
-            env = {}
+            env = None
             to_remove = None
         try:
             proc = subprocess.run(

--- a/marbles/core/tests/test_main.py
+++ b/marbles/core/tests/test_main.py
@@ -79,7 +79,7 @@ class CommandRunningTestCase(unittest.TestCase):
                 pth.flush()
 
             coverage_config = os.path.join(core_dir, '.coveragerc')
-            env = {**os.environ, **{'COVERAGE_PROCESS_START': coverage_config}}
+            env = {'COVERAGE_PROCESS_START': coverage_config, **os.environ}
             to_remove = pth.name
         else:
             env = None

